### PR TITLE
Add transaction guards management

### PR DESCRIPTION
**Pull request by germartinez**
## What it solves
It allows to add and remove Safe transaction guards

### CODE REVIEW

File: packages/safe-core-sdk/src/Safe.ts [hunk 1]
```diff
@@ -453,6 +466,49 @@ class Safe {
...
+   * @throws "Invalid guard address provided"
+   * @throws "There are no guards enabled already"
+   * @throws "Current version of the Safe does not support Safe transaction guards functionality"
+   */
+  async getDisableGuardTx(options?: SafeTransactionOptionalProps): Promise<SafeTransaction> {
+    const safeTransaction = await this.createTransaction({
+      to: this.getAddress(),
+      value: '0',
+      data: await this.#guardManager.encodeDisableGuardData(),
+      ...options
```
Comment (katspaugh) [inline packages/safe-core-sdk/src/Safe.ts:507]: Just to confirm, it's a conscious decision to allow the options to potentially override the three props above?

File: packages/safe-core-sdk/src/managers/guardManager.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  private validateGuardIsNotEnabled(currentGuard: string, newGuardAddress: string): void {
+    if (sameString(currentGuard, newGuardAddress)) {
+      throw new Error('Guard provided is already enabled')
+    }
+  }
+
+  private validateGuardIsEnabled(guardAddress: string): void {
+    if (isZeroAddress(guardAddress)) {
+      throw new Error('There are no guards enabled already')
```
Comment (katspaugh) [inline packages/safe-core-sdk/src/managers/guardManager.ts:30]: `already` -> `yet`

File: packages/safe-core-sdk/src/Safe.ts (hunk 1 above)
Comment (germartinez) [reply packages/safe-core-sdk/src/Safe.ts:529]: `to`, `value` and `data` are not included `options` type

File: packages/safe-core-sdk/src/Safe.ts [hunk 3]
```diff
@@ -475,6 +488,51 @@ class Safe {
...
+    }
+    const safeTransaction = await this.createTransaction({ safeTransactionData })
+    return safeTransaction
+  }
+
+  /**
+   * Returns the Safe transaction to disable a Safe guard.
+   *
+   * @param guardAddress - The desired guard address
```
Comment (iamacook) [inline packages/safe-core-sdk/src/Safe.ts:518]: I believe we can remove this.

File: packages/safe-core-sdk/src/managers/guardManager.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  private validateGuardIsEnabled(guardAddress: string): void {
+    if (isZeroAddress(guardAddress)) {
+      throw new Error('There are no guards enabled yet')
+    }
+  }
+
+  async getGuard(): Promise<string> {
+    // keccak256("guard_manager.guard.address")
+    const slot = '0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8'
```
Comment (iamacook) [inline packages/safe-core-sdk/src/managers/guardManager.ts:36]: Is it worthwhile moving this to a constant?

File: packages/safe-core-sdk/src/Safe.ts [hunk 5]
```diff
@@ -475,6 +488,51 @@ class Safe {
...
+   * @param options - The transaction optional properties
+   * @returns The Safe transaction ready to be signed
+   * @throws "Invalid guard address provided"
+   * @throws "There are no guards enabled already"
```
Comment (katspaugh) [inline packages/safe-core-sdk/src/Safe.ts:522]: Sorry, here as well.

**Pull request merged**